c/models/py_ftrl.cc

```diff
@@ -51,28 +52,30 @@ void Ftrl::m__init__(PKArgs& args) {
   const Arg& arg_lambda1          = args[3];
   const Arg& arg_lambda2          = args[4];
   const Arg& arg_nbins            = args[5];
-  const Arg& arg_nepochs          = args[6];
-  const Arg& arg_double_precision = args[7];
-  const Arg& arg_negative_class   = args[8];
+  const Arg& arg_mantissa_nbits   = args[6];
+  const Arg& arg_nepochs          = args[7];
+  const Arg& arg_double_precision = args[8];
+  const Arg& arg_negative_class   = args[9];
 
   bool defined_params           = !arg_params.is_none_or_undefined();
   bool defined_alpha            = !arg_alpha.is_none_or_undefined();
   bool defined_beta             = !arg_beta.is_none_or_undefined();
   bool defined_lambda1          = !arg_lambda1.is_none_or_undefined();
   bool defined_lambda2          = !arg_lambda2.is_none_or_undefined();
   bool defined_nbins            = !arg_nbins.is_none_or_undefined();
+  bool defined_mantissa_nbits      = !arg_mantissa_nbits.is_none_or_undefined();
   bool defined_nepochs          = !arg_nepochs.is_none_or_undefined();
   bool defined_double_precision = !arg_double_precision.is_none_or_undefined();
   bool defined_negative_class   = !arg_negative_class.is_none_or_undefined();
 
   if (defined_params) {
     if (defined_alpha || defined_beta || defined_lambda1 ||
-        defined_lambda2 || defined_nbins || defined_nepochs ||
-        defined_double_precision || defined_negative_class) {
+        defined_lambda2 || defined_nbins || defined_mantissa_nbits ||
+        defined_nepochs || defined_double_precision || defined_negative_class) {
 
       throw TypeError() << "You can either pass all the parameters with "
             << "`params` or any of the individual parameters with `alpha`, "
-            << "`beta`, `lambda1`, `lambda2`, `nbins`, `nepochs`, "
+            << "`beta`, `lambda1`, `lambda2`, `nbins`, `mantissa_nbits`, `nepochs`, "
```

Comment:
I wonder if we should perhaps allow this use case? `Ftrl(params=my_params, alpha=0.01)` would mean "use these parameters, but override alpha". Would be really easy to implement too: just delete "else" on line 112 below.

Reply:
This is very easy to be done, yes. Though I'm not sure it is a good idea to allow two `alpha`s to be passed, someone may do `Ftrl( alpha=0.01, params=my_params)` and easily be confused with what value is actually used :)

Reply:
Well, it's not actually that ambiguous: the interpretation where first `alpha=0.01` is applied and then overwritten with `params=` would mean the first alpha is completely useless. Whereas if you say `params=` is applied first and then `alpha=`, then the first params= is still useful: it sets up all other parameters.

Reply:
Well, we don't distinguish the order of `alpha` and `params`, so in both examples user ends up with the same set of parameters set for FTRL.  So it still looks ambiguous to me and may confuse people. What you say could already be achieved with just 

```
model = Ftrl(my_params)
model.alpha = 0.01
```

This looks much cleaner to me and doesn't require additional explanation.